### PR TITLE
AV-217108 Fix for Hostmap cache not cleared in delete case for Insecure EVH

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

File: internal/nodes/avi_model_evh_nodes.go [hunk 1]
```diff
@@ -2099,6 +2099,12 @@ func RouteIngrDeletePoolsByHostnameForEvh(routeIgrObj RouteIngressModel, namespa
 		}
 
 		SharedHostNameLister().DeleteNamespace(host)
+		if found, ingressHostMap := SharedHostNameLister().Get(host); found {
```
Comment (akshayhavile) [inline internal/nodes/avi_model_evh_nodes.go:2102]: @aaha97 Similar problem should exists for SNI insecure ingress deletion. Did we face issue over there?

File: internal/nodes/avi_model_evh_nodes.go [hunk 2]
```diff
@@ -2099,6 +2099,12 @@ func RouteIngrDeletePoolsByHostnameForEvh(routeIgrObj RouteIngressModel, namespa
...
+			mapkey := namespace + "/" + objname
+			if utils.HasElem(ingressHostMap.GetIngressesForHostName(), mapkey) {
+				delete(ingressHostMap.HostNameMap, mapkey)
+			}
```
Comment (akshayhavile) [inline internal/nodes/avi_model_evh_nodes.go:2106]: Once value is delete from map. We should save the updated map for that host.

File: internal/nodes/avi_model_evh_nodes.go (hunk 1 above)
Comment (aaha97) [reply internal/nodes/avi_model_evh_nodes.go:2102]: this was found when running evhtests. might find the issue in SNI UTs.

File: internal/nodes/avi_model_evh_nodes.go (hunk 2 above)
Comment (aaha97) [reply internal/nodes/avi_model_evh_nodes.go:2106]: this update to the map should be in-place as maps are passed by reference

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

File: internal/nodes/avi_model_evh_nodes.go (hunk 1 above)
Comment (aaha97) [reply internal/nodes/avi_model_evh_nodes.go:2102]: added the fix for SNI case

**Comment by aaha97** [comment]
build ako

File: internal/nodes/avi_model_evh_nodes.go (hunk 2 above)
Comment (akshayhavile) [reply internal/nodes/avi_model_evh_nodes.go:2106]: Do we need to guard delete of map with if condition?

**Comment by aaha97** [comment]
adding relevant UT results

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/ingresstests  1014.488s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests  1589.397s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests      1593.173s (vipPerNs)

File: internal/nodes/avi_model_evh_nodes.go (hunk 2 above)
Comment (aaha97) [reply internal/nodes/avi_model_evh_nodes.go]: not required, removed the if condition

**Comment by aaha97** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]
LGTM

**Comment by akshayhavile** [comment]
build ako

**Review by DixitAakash — APPROVED** [review]
LGTM

**Pull request merged**